apps/web/app/future/settings/(admin)/admin/apps/[category]/page.tsx

```diff
@@ -1,34 +1,52 @@
+import type { PageProps } from "app/_types";
 import { _generateMetadata } from "app/_utils";
 import { getFixedT } from "app/_utils";
 import { getServerSession } from "next-auth";
+import { notFound } from "next/navigation";
+import { z } from "zod";
 
 import { AUTH_OPTIONS } from "@calcom/feature-auth/lib/next-auth-options";
 import AdminAppsList from "@calcom/features/apps/AdminAppsList";
 import SettingsHeader from "@calcom/features/settings/appDir/SettingsHeader";
+import { AppRepository } from "@calcom/lib/server/repository/app";
+import { AppCategories } from "@calcom/prisma/enums";
 
 export const generateMetadata = async () =>
   await _generateMetadata(
     (t) => t("apps"),
     (t) => t("admin_apps_description")
   );
 
-const Page = async () => {
-  const session = await getServerSession(AUTH_OPTIONS);
+const querySchema = z.object({
+  category: z.nativeEnum(AppCategories).optional().default(AppCategories.calendar),
+});
 
+const Page = async ({ params, searchParams }: PageProps) => {
+  const session = await getServerSession(AUTH_OPTIONS);
   const t = await getFixedT(session?.user.locale || "en");
+  const { category } = querySchema.parse({ ...params, ...searchParams });
 
-  return (
-    <SettingsHeader title={t("apps")} description={t("admin_apps_description")}>
-      <div className="flex">
-        <AdminAppsList
-          baseURL="/settings/admin/apps"
-          classNames={{
-            appCategoryNavigationRoot: "overflow-x-scroll",
-          }}
-        />
-      </div>
-    </SettingsHeader>
-  );
+  try {
+    const appsList = await AppRepository.getAppsList({ input: { category } });
+
+    return (
+      <SettingsHeader title={t("apps")} description={t("admin_apps_description")}>
+        <div className="flex">
+          <AdminAppsList
+            baseURL="/settings/admin/apps"
+            classNames={{
+              appCategoryNavigationRoot: "overflow-x-scroll",
```

Comment:
@joeauyeung Hey Joe, please remind me. Did you add this? This is missing in /pages